Gems/Atom/Tools/MaterialCanvas/Assets/MaterialCanvas/TestData/asfloat.materialcanvasnode.azasset

```diff
@@ -0,0 +1,57 @@
+{
+    "Type": "JsonSerialization",
+    "Version": 1,
+    "ClassName": "DynamicNodeConfig",
+    "ClassData": {
+        "category": "Math Functions",
+        "title": "asfloat",
+        "inputSlots": [
+            {
+                "name": "inValue",
+                "displayName": "Value",
+                "description": "Value",
+                "supportedDataTypes": [
+                    "float",
```

Comment:
`asfloat` is a cast operation so shouldn't these input types be strictly integral types?

Reply:
I used code to quickly generate partial configurations for functions with simple signatures, requiring no more than 3 parameters, purely to have better coverage/variety for test data and fuel these kinds of conversations. That's why there are no descriptions and all the slot names are generic. I based the list off of HLSL documentation and excluded anything that I thought would not be supported yet. HLSL documentation says asfloat works with floating point and integer scalar and vector types. I left out the other conversions. I have not tested every function or combination. Will need to develop a strategy to do that before promoting anything as usable.

https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-asfloat